### PR TITLE
Update Assert module documentation

### PR DESCRIPTION
**Pull request by DanielBallaSZTE**
IoT.js-DCO-1.0-Signed-off-by: Daniel Balla dballa@inf.u-szeged.hu

### CODE REVIEW

File: docs/api/IoT.js-API-Assert.md [hunk 1]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
 | assert.fail | O | O | O |
 | assert.notEqual | O | O | O |
 | assert.notStrictEqual | O | O | O |
 | assert.strictEqual | O | O | O |
 | assert.throws | O | O | O |
 
+# Assert
 
-### Contents
+Assert module is for writing tests for applications.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Assert module is designed for

File: docs/api/IoT.js-API-Assert.md [hunk 2]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
+
+* `actual` {any} This property contains the actual value.
 
-Assert module is for writing test for applications.
+* `expected` {any} This property contains the expected value.
 
-You can get assert module by `require('assert')` the module provide.
+* `message` {any} The error message, default value is the error itself.
+
+* `name` {String} The name is set AssertionError.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: The name is `AssertionError` string.

File: docs/api/IoT.js-API-Assert.md [hunk 3]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
+
+* `operator` {String} This property contains the operator used for evaluating `actual` with `expected`.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: used for comparing `actual` with `expected`.

File: docs/api/IoT.js-API-Assert.md [hunk 4]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
 
-### `assert.assert(value[, message])`
-* `value <any>`
-* `message <any>`
+### assert(value[, message])
+* `value` {any} Value to test.
+* `message` {any} Message displayed in the thrown error.
 
-Test if `value` is true, it throws an `AssertionError` exception if `value` evaluates `false`.
+Checks if the `value` is truthy. If not, throws an AssertionError, with the given `message`.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: if the `value` is `true`. If it does not, the function throws
with the given optional `message`.

File: docs/api/IoT.js-API-Assert.md [hunk 5]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
+assert.assert(false, "it's false");
+// throws "AssertionError: it's false"
 ```
 
 
-### `assert.doesNotThrows(block[, message])`
-* `value <Function()>`
-* `message <any>`
+### doesNotThrow(block[, message])
+* `block` {Function} Expects a function.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Expects a function. - delete this

File: docs/api/IoT.js-API-Assert.md [hunk 6]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
+* `message` {any} Message to be displayed.
 
-Test if the given `block` does not throw any exception, if not, it throws an exception.
+Tests if the given `block` does not throw any exception, if it does, throws an
+exception with the given `message`.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: exception. If it does throws
with the given optional `message`.

File: docs/api/IoT.js-API-Assert.md [hunk 7]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
-* `actual <any>`
-* `expected <any>`
-* `message <any>`
+### equal(actual, expected[, message])
+* `actual` {any} The actual value.
+* `expected` {any} The expected value.
+* `message` {any} Message to be displayed.
 
-Test if `actual == expected` is evaluated to `true`, if not, it throws an exception.
+Tests if `actual == expected` is evaluated to `true`, if not, throws an
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: evaluated to `true`. If it does not, the function throws
given optional `message`.

File: docs/api/IoT.js-API-Assert.md [hunk 8]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
-* `message <any>`
-* `operator <String>`
+### fail(actual, expected, message, operator)
+* `actual` {any} The actual value.
+* `expected` {any} The expected value.
+* `message` {any} Message to be displayed.
+* `operator` {String} The operator.
 
-Throw an `AssertionError` exception.
+Throw an `AssertionError` exception with the given `message`.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Throws an ...

File: docs/api/IoT.js-API-Assert.md [hunk 9]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
-* `actual <any>`
-* `expected <any>`
-* `message <any>`
+### notEqual(actual, expected[, message])
+* `actual` {any} The actual value.
+* `expected` {any} The expected value.
+* `message` {any} Message to be displayed.
 
-Test if `actual != expected` is evaluated to `true`, if not, it throws an exception.
+Tests if `actual != expected` is evaluated to `true`, if not, throws an
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Please also fix these grammar errors.

File: docs/api/IoT.js-API-Assert.md (hunk 4 above)
Comment (DanielBallaSZTE) [inline docs/api/IoT.js-API-Assert.md]: I don't think there is a problem with truthy.
In JavaScript, a truthy value is a value that is considered  true when evaluated in a Boolean context.
> If it does not

`If it is not` would be the right phrase, as you are referring to the value, not the check itself.

**Comment by DanielBallaSZTE** [comment]
Updated the request, adding those issued changes.

File: docs/api/IoT.js-API-Assert.md [hunk 10]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
+
+* `actual` {any} This property contains the actual value.
 
-Assert module is for writing test for applications.
+* `expected` {any} This property contains the expected value.
 
-You can get assert module by `require('assert')` the module provide.
+* `message` {any} The error message, default value is the error itself.
+
+* `name` {String} The name is `AssertionError` string.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: I think there is no need for newline separaters between fields.

File: docs/api/IoT.js-API-Assert.md [hunk 11]
```diff
@@ -5,203 +5,234 @@ The following shows Assert module APIs available for each platform.
...
-* `actual <any>`
-* `expected <any>`
-* `message <any>`
+### notEqual(actual, expected[, message])
+* `actual` {any} The actual value.
+* `expected` {any} The expected value.
+* `message` {any} Message to be displayed.
 
-Test if `actual != expected` is evaluated to `true`, if not, it throws an exception.
+Tests if `actual != expected` is evaluated to `true`, if it is not, throws an
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: I like shorter sentences, but this is just my preference. I leave this to the other reviewers.

Tests if `actual != expected` is evaluated to `true`. If it is not throws an...

**Comment by DanielBallaSZTE** [comment]
@zherczeg removed the new lines from there.

File: docs/api/IoT.js-API-Assert.md [hunk 12]
```diff
@@ -5,203 +5,230 @@ The following shows Assert module APIs available for each platform.
...
 
 
-# Assert
+## Class: AssertionError
+
+Assert module will produce `AssertionError` in case of an assertion failure. `AssertionError` inherits standard `Error` thus it has properties provided by `Error` object including additional properties.
+
 
-Assert module is for writing test for applications.
+### Properties
```
Comment (daeyeon) [inline docs/api/IoT.js-API-Assert.md]: Please check #873 - 6. Remove all the section titles such as Class Methods, Prototype Function, Module Functions/Properties, Properties, Event, and so on.

Please refer to https://github.com/bzsolt/iotjs/blob/ac74208d8c2cd1453d8ec7bc561cf48c687137fc/docs/api/IoT.js-API-HTTP.md#messagestatuscode for Properites.

File: docs/api/IoT.js-API-Assert.md [hunk 13]
```diff
@@ -5,203 +5,230 @@ The following shows Assert module APIs available for each platform.
...
-You can get assert module by `require('assert')` the module provide.
+
+* `actual` {any} This property contains the actual value.
+* `expected` {any} This property contains the expected value.
+* `message` {any} The error message, default value is the error itself.
+* `name` {String} The name is `AssertionError` string.
+* `operator` {String} This property contains the operator used for comparing `actual` with `expected`.
 
 
 ## Module Functions
```
Comment (daeyeon) [inline docs/api/IoT.js-API-Assert.md]: Please check #873

**Comment by DanielBallaSZTE** [comment]
I've fixed those issues. Or did you mean to name all the methods as assert.something?

File: docs/api/IoT.js-API-Assert.md [hunk 14]
```diff
@@ -5,203 +5,227 @@ The following shows Assert module APIs available for each platform.
...
 
-# Assert
+## Class: AssertionError
+
+Assert module will produce `AssertionError` in case of an assertion failure. `AssertionError` inherits standard `Error` thus it has properties provided by `Error` object including additional properties.
 
-Assert module is for writing test for applications.
 
-You can get assert module by `require('assert')` the module provide.
+### Properties
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Remove Properties line.

Comment (DanielBallaSZTE) [inline docs/api/IoT.js-API-Assert.md]: Fixed that too

**Comment by DanielBallaSZTE** [comment]
Added minor styling enhancements.

File: docs/api/IoT.js-API-Assert.md [hunk 15]
```diff
@@ -5,203 +5,224 @@ The following shows Assert module APIs available for each platform.
...
-* `actual <any>`
-* `expected <any>`
-* `message <any>`
+### equal(actual, expected[, message])
+* `actual` {any} The actual value.
+* `expected` {any} The expected value.
+* `message` {any} Message to be displayed.
 
-Test if `actual == expected` is evaluated to `true`, if not, it throws an exception.
+Tests if `actual == expected` is evaluated to `true`. Elseways throws an
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Elseways -> Otherwise

File: docs/api/IoT.js-API-Assert.md [hunk 16]
```diff
@@ -5,203 +5,224 @@ The following shows Assert module APIs available for each platform.
...
-* `actual <any>`
-* `expected <any>`
-* `message <any>`
+### notStrictEqual(actual, expected[, message])
+* `actual` {any} The actual value.
+* `expected` {any} The expected value.
+* `message` {any} Message to be displayed.
 
-Test if `actual !== expected` is evaluated to `true`, if not, it throws an exception.
+Tests if `actual !== expected` is evaluated to `true`. Elseways throws an exception
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Ditto.

File: docs/api/IoT.js-API-Assert.md [hunk 17]
```diff
@@ -5,203 +5,224 @@ The following shows Assert module APIs available for each platform.
...
-* `actual <any>`
-* `expected <any>`
-* `message <any>`
+### throws(block[, expected, message])
+* `block` {Function} The function that throws an error.
+* `expected` {Function} The expected error type.
+* `message` {any} Message to be displayed.
 
-Test if the given `block` throws an `expected` error, if not, it throws an exception.
+Tests if the given `block` throws an `expected` error. Elseways throws an exception
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Assert.md]: Ditto.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**